src/main/badge.ts

```diff
@@ -2,20 +2,79 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {app} from 'electron';
+import {BrowserWindow, app, nativeImage} from 'electron';
 import log from 'electron-log';
 
 import {UPDATE_BADGE} from 'common/communication';
 
 import {localizeMessage} from 'main/i18nManager';
 
-import WindowManager from './windows/windowManager';
 import * as AppState from './appState';
+import MainWindow from './windows/mainWindow';
 
 const MAX_WIN_COUNT = 99;
 
 let showUnreadBadgeSetting: boolean;
 
+/**
+     * Badge generation for Windows
+     */
+
+function drawBadge(text: string, small: boolean) {
+    const scale = 2; // should rely display dpi
+    const size = (small ? 20 : 16) * scale;
+    const canvas = document.createElement('canvas');
+    canvas.setAttribute('width', `${size}`);
+    canvas.setAttribute('height', `${size}`);
+    const ctx = canvas.getContext('2d');
+
+    if (!ctx) {
+        log.error('Could not create canvas context');
```

Comment:
should we log the reason for this or is there a particular reason that we already know this happens?

Reply:
Not sure, I'm guessing it's there since it's possible `canvas.getContext` can return null (probably if you provide the wrong string or something) and if it does we don't want to proceed. I think we log it in case someone asks why the badge isn't there and if so we can point to that code.